cmd/serve/main.go

```diff
@@ -8,7 +8,10 @@ import (
 
 func main() {
 	srv := server.New(server.NewConfig())
-	srv.RegisterResource(routes.NewResource("/health"))
+
+	healthResource := routes.HealthResource{}
```

Comment:
@damiannolan any input here? I know go is ment to be functional and this is OO, but I did it this way so we can have multiple file in the `/routes` folder.

Wanted to keep the folder structure neater.

Reply:
Totally happy to go with this too if you want ya. I do think tho, that we should abstract the "service" logic if you will away from the http handlers. It would follow the idiomatic go convention pretty well and we'd also be able to further down the line do nice stuff like for example - use Cobra (the vip3r sister lib) for making CLI apps which behave the exact same as the http interface.

Reply:
This is an awesome idea. I have a first draft of what I think this looks like in the PR below. ✊

https://github.com/blockchain-abstraction-middleware/project-template/pull/3/files#diff-e1db7291339125bc61ac619596551ea4R10

wdyt?